lib/err.c

```diff
@@ -0,0 +1,97 @@
+/*
+** Copyright (C) 2020 University of Oxford
+**
+** This file is part of tsinfer.
+**
+** tsinfer is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** tsinfer is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with tsinfer.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "err.h"
+
+const char *
+tsi_strerror(int err)
+{
+    const char *ret = "Unknown error";
+
+    switch (err) {
+        case 0:
+            ret = "Normal exit condition. This is not an error!";
+            break;
+
+        case TSI_ERR_GENERIC:
+            ret = "Generic error - please file a bug report.";
```

Comment:
Maybe "Generic tsinfer error"? Or is that context always clear in the stacktrace?

Reply:
Done.